secant/Features/BackupFlow/DropDelegate.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  DropDelegate.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 11/16/21.
+//
+
+import Foundation
+import SwiftUI
+import OrderedCollections
+import ComposableArchitecture
+
+/// Drop delegate that accepts items conforming to `PhraseChip.validationWordTypeIdentifier`
+struct WordChipDropDelegate: DropDelegate {
+    var dropAction: ((PhraseChip.Kind) -> Void)?
+
+    func validateDrop(info: DropInfo) -> Bool {
```

Comment:
Is this used anywhere?

Reply:
the `validateDrop`? or the whole delegate?
This is used by the drag and drop API from Swift UI (well actually UIKit through SwiftUI) to perform the drop

Reply:
````Swift
 .whenIsDroppable(
                                !state.groupCompleted(index: index),
                                dropDelegate: WordChipDropDelegate { chipKind in
                                    viewStore.send(.move(wordChip: chipKind, intoGroup: index))
                                }
                            )
````

Reply:
then

````Swift
extension View {
    /// Makes a View accept drop types Self.validationWordTypeIdentifier when it is of kind .empty
    func whenIsDroppable(_ isDroppable: Bool, dropDelegate: DropDelegate) -> some View {
        self.modifier(MakeDroppableModifier(isDroppable: isDroppable, drop: dropDelegate))
    }
}

struct MakeDroppableModifier: ViewModifier {
    var isDroppable: Bool
    var drop: DropDelegate
    
    func body(content: Content) -> some View {
        if isDroppable {
            content.onDrop(of: [PhraseChip.validationWordTypeIdentifier], delegate: drop)
        } else {
            content
        }
    }
}
````

Reply:
I misunderstood, I see now!